### PR TITLE
Remove Defined markers in simplifyConjunctionByAssumption

### PR DESCRIPTION
**Pull request by ana-pantilie**
~Needs #2366~

Until #2351 is done, this will fix those cases where a conjunction isn't simplified correctly because the terms differ by `Defined` markers.

---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [ ] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [ ] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [ ] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [ ] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

### CODE REVIEW

File: kore/test/Test/Kore/Step/Simplification/Condition.hs [hunk 1]
```diff
@@ -76,13 +76,18 @@ test_simplify_local_functions =
     , test "contradiction: \"one\" = f(x) ∧ f(x) = \"two\"" fString (Left  strOne) (Right strTwo)
     , test "contradiction: \"one\" = f(x) ∧ \"two\" = f(x)" fString (Left  strOne) (Left  strTwo)
     , test "contradiction: f(x) = \"one\" ∧ \"two\" = f(x)" fString (Right strOne) (Left  strTwo)
+    -- Ignore Defined marker
+    , testDefined "contradiction: f(g(x)) = a ∧ f(g(x)) = b" (Right fg) (Right a) (Right b)
+    , testDefined "contradiction: f(g(x)) = a ∧ f(g(x)) = b" (Left fg) (Right a) (Right b)
```
Comment (ttuegel) [inline kore/test/Test/Kore/Step/Simplification/Condition.hs:81]: What is the difference between `Right fg` and `Left fg`?

File: kore/test/Test/Kore/Step/Simplification/Condition.hs [hunk 2]
```diff
@@ -110,6 +120,18 @@ test_simplify_local_functions =
...
 
+    testDefined name eitherFunc eitherC1 eitherC2 =
+        testCase name $ do
+            let equals1 =
+                    mkLocalDefn (applyDefined1 eitherFunc) eitherC1
+                        & Condition.fromPredicate
+                equals2 =
+                    mkLocalDefn (applyDefined2 eitherFunc) eitherC2
+                        & Condition.fromPredicate
+                condition = definedFg <> equals1 <> definedFg <> equals2
```
Comment (ttuegel) [inline kore/test/Test/Kore/Step/Simplification/Condition.hs:131]: Using `definedFg` here actually requires that `eitherFunc` is `Right fg` or `Left fg`, so I would rather we not allow passing in any value there.

**Review by ttuegel — CHANGES_REQUESTED** [review]

File: kore/test/Test/Kore/Step/Simplification/Condition.hs (hunk 1 above)
Comment (ana-pantilie) [reply kore/test/Test/Kore/Step/Simplification/Condition.hs:81]: `Right fg` will apply `mkDefined` to the `f(g(x)` in the second equals, and `Left fg` to the one in the first.

File: kore/test/Test/Kore/Step/Simplification/Condition.hs (hunk 2 above)
Comment (ana-pantilie) [reply kore/test/Test/Kore/Step/Simplification/Condition.hs:131]: Do you mean we should remove `definedFg` from `condition`?

**Review by ttuegel — APPROVED** [review]

**Pull request merged**